src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -932,6 +932,13 @@ ProcessCLRExceptionNew(IN     PEXCEPTION_RECORD   pExceptionRecord,
 
     Thread* pThread         = GetThread();
 
+    if (pExceptionRecord->ExceptionCode == STATUS_LONGJUMP)
+    {
+        // This is an exception used to unwind during longjmp function. We just let it pass through managed
```

Comment:
Do we need to unwind Frames to avoid corrupting runtime data structures?

Reply:
And if we are unwinding Frames, is it that hard to call finallys too to match the old behavior?

Reply:
I had it implemented by propagating the exception through the managed frames as usual, so the finallys were called, and then re-issued the longjmp after processing them. The only missing part to do was preventing calling catch handlers. The old EH doesn't call those "naturally", as the longjmp runs only the unwind pass.
And then you've mentioned it was not supported, so I have deleted that code :-).
But anyways, I can change it back to that, the change wasn't complicated. The only thing is that the Unix handling would then be different.

Reply:
> The only thing is that the Unix handling would then be different.

longjump over managed code does not and cannot work well on Unix, so I do not think it is a problem